plugins/shortcuts/src/ShortcutItem.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { SidebarItem } from '@backstage/core';
+import { IconButton, makeStyles } from '@material-ui/core';
+import EditIcon from '@material-ui/icons/Edit';
+import { ShortcutIcon } from './ShortcutIcon';
+import { EditShortcut } from './EditShortcut';
+import { ShortcutApi } from './api';
+import { Shortcut } from './types';
+
+const useStyles = makeStyles({
+  icon: {
+    color: 'white',
+    fontSize: 16,
+  },
+});
+
+const getIconText = (title: string) =>
+  title.split(' ').length === 1
+    ? // If there's only one word, keep the first two characters
+      title[0].toUpperCase() + title[1].toLowerCase()
+    : // If there's more than one word, take the first character of the first two words
+      title
+        .replace(/\W+/g, ' ')
+        .split(' ')
+        .map(s => s[0])
+        .join('')
+        .slice(0, 2)
+        .toUpperCase();
```

Comment:
Just to note, in `@backstage/core` there's an `<OverflowToolTip>` component which also uses a react component called `<TextTruncate>`.  Not sure if there's some over lap or reuse possible there to eliminate some of this mapping/splitting, etc, but just wanted to point it out!

Reply:
Thanks! I didn't now about that. But after taking a quick look I don't think I can use it the way I want.